ApeWorX/ape
# Feat/network cmd

### What I did

Made the network context manager happen automatically by either using a `network_bound_command` decorator
You can also use the custom `ape_group` decorator on a `click.group` and then from there, do `cli.network_bound_command()`.

### How I did it

First, I noticed that this didn't happen automatically, and I wanted it to, for ease of development.
Next, I created the custom `NetworkBoundCommand`
Then I made it accessible as a decorator, including off a custom group so it can be used either in command or group-command.

### How to verify it

Add
`print(networks.active_provider.network.name)` to the run or console commands and set `--network` and see that it gets sets.

Also, try this:

```
from ape.cli import ape_group, network_option

@ape_group()
def cli():
    """My custom Ethereum tools"""
   
@cli.network_bound_command()
@network_option
def cmd(network):
  print(networks.active_provider.network.name)
```

### Checklist
